### DISCUSSION TITLE
Does the industrial core support the real-time trajectory.

### DISCUSSION THREAD

**Discussion by caimingxue** [ros-industrial/industrial_core#283] [Q&A]
That is to say, I use the moveit servo to generate real-time joint trajectory ( every trajectory only contains one point), the industrial core can receive this kind of joint trajectory and send to the real robot according to simple message protocol?

**Comment by gavanderhoorn** [comment]
Technically, this would be possible.

Realistically though: most (all?) drivers I know of that use `industrial_robot_client` in some way only really support the `FollowJointTrajectoryAction` server interface.

Sending single-point trajectories to that will not result in smooth motion across different trajectories -- as the implementation in `industrial_robot_client` does not support trajectory replacement.

There are some extensions which have implemented something like a *streaming interface* (see https://github.com/ros-industrial/motoman/pull/215 fi). Those are not part of the packages maintained here though, so we can't say how they would behave with something like `moveit_servo`.

In my experience, `moveit_servo` only really works well when it has access to a (moderately) high-bandwidth, low-latency, low-jitter external motion interface -- ideally with some on-OEM-controller filtering.

For robots which do not have such an interface it can be difficult to get it to work properly.
